source/transfer_list.rst

```diff
@@ -657,5 +660,120 @@ such that the last ACPI table in this entry ends at offset
      - hdr_size
      - One or more ACPI tables.
 
+.. _optee_pagable_part_entry:
+
+OP-TEE pagable part entry layout (XFERLIST_OPTEE_PAGABLE_PART)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This entry type holds the address of OP-TEE pagable part.
+This address (of type 'uint64_t') is used when OPTEED(OP-TEE Dispatcher)
+is the Secure Payload Dispatcher, indicating where to load the pagable image of
+OP-TEE OS.
+
+.. _tab_optee_pagable_part:
+.. list-table:: OP-TEE pagable part type layout
+   :widths: 2 2 2 8
+
+   * - Field
+     - Size (bytes)
+     - Offset (bytes)
+     - Description
+
+   * - tag_id
+     - 0x3
+     - 0x0
+     - The tag_id field must be set to **0x20**.
+
+   * - hdr_size
+     - 0x1
+     - 0x3
+     - |hdr_size_desc|
+
+   * - data_size
+     - 0x4
+     - 0x4
+     - The size of OP-TEE pagable part in bytes must be set to **8**.
+
+   * - pagable_part
+     - data_size
+     - hdr_size
+     - Pagable part address for OPTEED
+
+.. _tos_fw_config_entry:
+
+TOS FW config entry layout (XFERLIST_TOS_FW_CONFIG)
```

Comment:
TOS_FW_CONFIG is a TF-A concept intended to contain all the config data for a TOS. But it seems you're using it here to specifically contain the SPMC manifest, which is a standard FF-A spec structure, not specific to OP-TEE. I think you should create an SPMC manifest TE, which refers to the FF-A spec. This can have a top level tag_id instead of being scoped to be in the OP-TEE space.

Reply:
Sounds good, but now all OP-TEE specific TEs are starting from 0x20, if we use it for a general purpose to refer to FF-A, I will move it to the general ones staring from 0. Is it good for you?

Reply:
Sounds good, although we have other in-flight PRs that add new TEs so we might have change the number depending on which is merged first.

Reply:
Done.

Reply:
Wouldn't it be best to have a separate "Arm" or "FF-A" block of tag IDs for this then? If this specification is truly supposed to be universal (e.g. include x86 PCs), then I wouldn't expect FF-A specific definitions to go into the "general" block. (Especially if we only leave 32 IDs total for that, I think that's a bad idea in general, I would at least leave 256 and start the next block at 0x100...)

Reply:
A once allocated TE may never be removed (at most it may be marked "deprecated"), so it would be better to try to make a final decision here right now.

Reply:
Understand.
Then I will prefer to @jwerner-chromium 's suggestion, move OP-TEE TEs to start from 0x100 and keep all Arm/FF-A related TEs as "generic" ones start from 0.
Is this fine for you? @sjg20 @danh-arm

Reply:
What about we keep truly generic ones like DTB/HOBLIST/TPM starting at 0x0 and let all Arm specific once starting 0x100. Arm specific once can include all the projects which are hosted in tf.org (TF-A/OP-TEE/FFA)

Reply:
> What about we keep truly generic ones like DTB/HOBLIST/TPM starting at 0x0 and let all Arm specific once starting 0x100. Arm specific once can include all the projects which are hosted in tf.org (TF-A/OP-TEE/FFA)

This is fine for me. @sjg20 @danh-arm @jwerner-chromium , is this good for you? If all of you agree I will amend the PR.

Reply:
How many Arm-specific ones are we expecting to see there? I would like avoid a very sparse list. So long as expect several dozen over time, then that seems OK to me

Reply:
Thanks @raymo200915 !
To understand the usage of XFERLIST_OPTEE_PAGABLE_PART i was looking at your patch [1], this tag basically passes a base address and limit for the next stage for its usage. There is very similar usecase in bl1/bl2 interface where BL1 passes memory layout for BL2. Which basically a structure like below 
typedef struct meminfo {
        uintptr_t total_base;
        size_t total_size;
} meminfo_t;

I was just wondering if we could have a more generic name for this entry (XFERLIST_MEM_INFO or XFERLIST_MEM_LAYOUT) which can be used by receiving stage for its memory usage. This can even go in the generic range.

[1] https://review.trustedfirmware.org/c/TF-A/trusted-firmware-a/+/23777

Reply:
@manish-pandey-arm
`XFERLIST_OPTEE_PAGABLE_PART` is passing the 'base address' of the OP-TEE pagable part only, not the memory range pointed by that address.
Other than that, if we use a generic name, then there might be multiple TEs with same tag even within one boot stage, how OP-TEE knows which one  is containing the pagable part to consume?

Reply:
Yes I think it is good to have this in its own TE, so I don't agree with using generic mem info/layout for this.

But can you explain why the size is not required?

Reply:
Ack, for having its own TE.